File: 1838.frequency-of-the-most-frequent-element.java
import java.util.*;
class Solution {
    	public int maxFrequency(int[] nums, int k) 
        {
            Arrays.sort(nums);
            int r = 0, l = 0, sum = 0, max_val = 0;
            for (r=0;r<nums.length;r++)
            {
                sum += nums[r];
                while (nums[r] * (r - l + 1) - sum > k){ sum -= nums[l++];}
                max_val = Math.max(max_val,r-l+1);
            }
        return max_val;
    }
}

File: 1838-frequency-of-the-most-frequent-element/1838-frequency-of-the-most-frequent-element.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        l,r,total,solution= 0,0,0,0
        while(r < len(nums)):
            total += nums[r]
            while(nums[r]*(r-l+1) > total+k):
                total -= nums[l]
                l += 1
            solution = max(solution,r-l+1)
            r += 1
        return solution